documentation/specs/CodeVisualizer.md

```diff
@@ -0,0 +1,135 @@
+# `CodeVisualizer` Component Specification
+
+## Design
+
+`CodeVisualizer` wraps one or more `CodeBlock`s with a header to allow the user to switch between blocks.
+
+`CodeVisualizer` will be a controlled component. It will require the consumer to provide a selected language and manage its state.
+
+### API
+
+```ts
+export type CodeVisualizerProps = {
+  /**
+   * CodeVisualizer content. This should be a header and collection of snippets.
+   */
+  children: ReactNode;
+
+  /**
+   * Selected language.
+   */
+  language: SnippetLanguages;
+
+  /**
+   * Callback for when the selected language changes.
+   */
+  onLanguageChange: (language: SnippetLanguages) => void;
+};
+
+export type CodeVisualizerHeaderProps = {
+  /**
+   * Header title.
+   */
+  children: ReactNode;
+
+  /**
+   * Header color.
+   *
+   * @default neutral
+   */
+  color?: "neutral" | "primary" | "secondary";
+};
+
+export type CodeVisualizerSnippetProps = {
+  /**
+   * Code snippet content. This should contain a CodeBlock.
+   */
+  children: ReactNode;
+};
+```
+
+### Example Usage
+
+```tsx
+import { CodeVisualizer } from "@easypost/easy-ui/CodeBlock";
```

Comment:
**question**: is the language picker/selector intended to function as a trigger for a menu/listbox? if so, do you have a rough idea for how you want to implement the trigger?

Reply:
yeah it's supposed to trigger our `Menu`. i was imagining using an `UnstyledButton` with a `Text` and `Icon` inside as the `Menu.Trigger`